google/api_core/retry.py

```diff
@@ -201,7 +202,12 @@ def retry_target(
 
     for sleep in sleep_generator:
         try:
-            return target()
+            result = target()
+            if inspect.isawaitable(result):
+                raise exceptions.GoogleAPICallError(
+                    "Warning: Use google.api_core.retry_async.AsyncRetry for async calls!"
```

Comment:
Consider changing the wording to something like
"Warning: Using the synchronous google.api_core.retry.Retry with asynchronmous calls may lead to unexpected results. Please use google.api_core.retry_async.AsyncRetry instead."

Motivation: we summarize the problem for the user, giving context for our warning/recommendation.